examples/memory_with_compression/_memory_with_compress.py

```diff
@@ -0,0 +1,587 @@
+# -*- coding: utf-8 -*-
+"""The MemoryWithCompress class for memory management with compression."""
+
+import asyncio
+import concurrent.futures
+import copy
+import json
+from typing import (
+    Any,
+    Awaitable,
+    Callable,
+    Iterable,
+    List,
+    Optional,
+    Sequence,
+    Union,
+)
+
+from agentscope.formatter import FormatterBase
+from agentscope.memory import MemoryBase
+from agentscope.message import Msg
+from agentscope.model import ChatModelBase
+from agentscope.token import TokenCounterBase
+
+# pylint: disable=wrong-import-order
+from _mc_utils import (  # noqa: E402
+    count_words,
+    format_msgs,
+    DEFAULT_COMPRESSION_PROMPT_TEMPLATE,
+    MemoryCompressionSchema,
+)
+from _memory_storage import (  # noqa: E402
+    InMemoryMessageStorage,
+    MessageStorageBase,
+)
+
+
+class MemoryWithCompress(MemoryBase):
+    """
+    MemoryWithCompress is a memory manager that stores original messages
+    in chat_history_storage and compressed messages in memory_storage.
+    """
+
+    def __init__(
+        self,
+        model: ChatModelBase,
+        formatter: FormatterBase,
+        max_token: int = 28000,
+        chat_history_storage: MessageStorageBase = InMemoryMessageStorage(),
+        memory_storage: MessageStorageBase = InMemoryMessageStorage(),
+        token_counter: Optional[TokenCounterBase] = None,
+        compress_func: Callable[[List[Msg]], Awaitable[List[Msg]]]
+        | None = None,
+        compression_trigger_func: Callable[[List[Msg]], Awaitable[bool]]
+        | None = None,
+        compression_on_add: bool = False,
+        compression_on_get: bool = True,
+        customized_compression_prompt: str | None = None,
+    ) -> None:
+        """Initialize the MemoryWithCompress.
+
+        Args:
```

Comment:
missing `chat_history_storage` & `memory_storage`

Reply:
fix; add the description in args